rpcs3/Emu/Cell/lv2/sys_rsx.cpp

```diff
@@ -253,13 +281,13 @@ s32 sys_rsx_context_attribute(s32 context_id, u32 package_id, u64 a3, u64 a4, u6
 
 	auto m_sysrsx = fxm::get<SysRsxConfig>();
 
-	if (!m_sysrsx)
+	if (!m_sysrsx || !m_sysrsx->context_base)
 	{
 		sys_rsx.error("sys_rsx_context_attribute called before sys_rsx_context_allocate: context_id=0x%x, package_id=0x%x, a3=0x%llx, a4=0x%llx, a5=0x%llx, a6=0x%llx)", context_id, package_id, a3, a4, a5, a6);
 		return CELL_EINVAL;
 	}
 
-	auto &driverInfo = vm::_ref<RsxDriverInfo>(m_sysrsx->driverInfo);
+	auto &driverInfo = vm::_ref<RsxDriverInfo>(m_sysrsx->context_base + 0x100000);
```

Comment:
So the driverInfo member was removed and replaced with raw address casts? Why? Is there some kind of restriction to make m_sysrsx have only 3 members? The removal of this field is certainly strange.

Reply:
rsx_device_map and rsx_allocate_context can be called in any order, in each order they will have a different address. you can calculate driverinfo by adding 1m to context base so it can be removed.

Reply:
Isn't context base only set in one method though? That still does not make sense here. Is there another method that alters context_base?

Reply:
Im not sure what about what are you asking, are you referring to driverinfo removal or the addition of device_addr? either way, context base is only set once.

Reply:
I can readd driverinfo if it looks better, it doesnt hurt anything.